json/access/src/main/java/ch/qos/logback/contrib/json/access/JsonLayout.java

```diff
@@ -0,0 +1,465 @@
+ *         <td nowrap="nowrap"><code>IAccessEvent.{@link ch.qos.logback.access.spi.IAccessEvent#getResponseContent() getResponseContent()}</code></td>
+ *         <td></td>
+ *         <td>false</td>
+ *     </tr>
+ * </table>
+ * <p/>
+ *
+ * @author Espen A. Fossen
+ */
+public class JsonLayout extends JsonLayoutBase<IAccessEvent> {
+
+    public static final String TIMESTAMP_ATTR_NAME = "timestamp";
+    public static final String REMOTEADDR_ATTR_NAME = "remoteAddress";
+    public static final String REMOTEUSER_ATTR_NAME = "remoteUser";
+    public static final String REQUESTTIME_ATTR_NAME = "requestTime";
+    public static final String REQUESTURI_ATTR_NAME = "uri";
+    public static final String STATUSCODE_ATTR_NAME = "status";
+    public static final String METHOD_ATTR_NAME = "method";
+    public static final String PROTOCOL_ATTR_NAME = "protocol";
+    public static final String CONTENTLENGTH_ATTR_NAME = "contentLength";
+    public static final String REQUESTURL_ATTR_NAME = "url";
+    public static final String REMOTEHOST_ATTR_NAME = "remoteHost";
+    public static final String SERVERNAME_ATTR_NAME = "serverName";
+    public static final String REQUESTHEADER_ATTR_NAME = "headers";
+    public static final String REQUESTPARAMETER_ATTR_NAME = "params";
+    public static final String LOCALPORT_ATTR_NAME = "port";
+    public static final String REQUESTCONTENT_ATTR_NAME = "requestContent";
+    public static final String RESPONSECONTENT_ATTR_NAME = "responseContent";
+
+    protected boolean includeRemoteAddr;
+    protected boolean includeRemoteUser;
+    protected boolean includeRequestTime;
+    protected boolean includeRequestURI;
+    protected boolean includeStatusCode;
+    protected boolean includeMethod;
+    protected boolean includeProtocol;
+    protected boolean includeContentLength;
+    protected boolean includeRequestURL;
+    protected boolean includeRemoteHost;
+    protected boolean includeServerName;
+    protected boolean includeRequestHeader;
+    protected boolean includeRequestParameter;
+    protected boolean includeLocalPort;
+    protected boolean includeRequestContent;
+    protected boolean includeResponseContent;
+
+
+    public JsonLayout() {
+        super();
+        this.includeRemoteAddr = true;
+        this.includeRemoteUser = true;
+        this.includeRequestTime = true;
+        this.includeRequestURI = true;
+        this.includeStatusCode = true;
+        this.includeMethod = true;
+        this.includeProtocol = true;
+        this.includeRequestURL = false;
+        this.includeContentLength = false;
+        this.includeRemoteHost = true;
+        this.includeServerName = true;
+        this.includeRequestHeader = true;
+        this.includeRequestParameter = true;
+        this.includeLocalPort = false;
+        this.includeRequestContent = false;
+        this.includeResponseContent = false;
+    }
+
+    @Override
+    protected Map toJsonMap(IAccessEvent event) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method toJsonMap() has an NPath complexity of 510183360](https://www.codacy.com/app/tony19/logback-contrib/file/2370841728/issues/source?bid=3222643&fileBranchId=3222645#l201)